File: webflux-error-list/src/main/java/rodrigues/adriane/demo/webfluxerrorlist/ContactService.java
package rodrigues.adriane.demo.webfluxerrorlist;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;
import reactor.core.publisher.Mono;
import rodrigues.adriane.demo.webfluxerrorlist.entity.Contact;

@Slf4j
@Service
@RequiredArgsConstructor
public class ContactService {

    private final ContactRepository contactRepository;

    public Mono<Contact> saveContact(final Contact contact) {
        log.info("save contact {}", contact);
        return contactRepository.save(contact);
    }

    public Mono<Contact> findContact(final Integer id) {
        return contactRepository.findById(id);
    }
}

File: jax-rs-demo/src/main/kotlin/br/com/adriane/demo/jaxrs/service/ContactService.kt
package br.com.adriane.demo.jaxrs.service

import br.com.adriane.demo.jaxrs.entity.Contact
import br.com.adriane.demo.jaxrs.repository.ContactRepository
import br.com.adriane.demo.jaxrs.rest.dto.ContactDto
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Service

@Service
class ContactService(
        private val contactRepository: ContactRepository
) {
    private val log = LoggerFactory.getLogger(ContactService::class.java)

    fun insertContact(dto: ContactDto): ContactDto {
        return mapEntityToDto(
                contactRepository.save(
                        mapDtoToEntity(dto)
                )
        )
    }

    fun listContacts(): List<ContactDto> =
            contactRepository.findAll().map(this::mapEntityToDto)

    private fun mapDtoToEntity(dto: ContactDto): Contact =
            Contact(dto.id, dto.name, dto.email)

    private fun mapEntityToDto(contact: Contact): ContactDto =
            ContactDto(contact.id, contact.name, contact.email)
}